docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -340,6 +342,60 @@ You will also need to configure the `maven-compiler-plugin` to detect the `quark
 </build>
 ----
 
+==== Create new extension modules using Maven
+
+The `create-extension` mojo of `io.quarkus:quarkus-maven-plugin` can be used to generate stubs of Maven modules needed for implementing a new Quarkus extension.
+
+WARNING: This mojo can be currently used only for adding extensions to an established source tree, such as https://github.com/quarkusio/quarkus[Quarkus] or https://github.com/quarkusio/quarkus[Camel Quarkus]. Creating extension projects from scratch is not supported yet.
+
+As and example, let's add a new extension called `my-ext` to the Quarkus source tree:
+
+[source, shell, subs=attributes+]
+----
+git clone https://github.com/quarkusio/quarkus.git
+cd quarkus
+cd extensions
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create-extension -N -Dquarkus.artifactIdBase=my-ext
+----
+
+The above sequence of commands will create three new Maven modules:
+
+* `quarkus-my-ext-parent` in the `my-ext` directory; this module is also added to `<modules>` of `quarkus-extensions-parent`
+* `quarkus-my-ext` in the `my-ext/runtime` directory
+* `quarkus-my-ext-deployment` in the `my-ext/deployment` directory; a basic `MyExtProcessor` class is generated in this module.
+
+The newly added modules are not suitable for building immediatelly.
+You will need to perform some adjustments manually (some of them are specific to the Quarkus source tree):
+
+* Add `quarkus-my-ext` to the `dependencyManagement` in `bom/runtime/pom.xml`
+* Add `quarkus-my-ext-deployment` to the `dependencyManagement` in `bom/deployment/pom.xml`
+* Add the new extension coordinates to `devtools/common/src/main/filtered/extensions.json`
+
+Note that the parameters of the mojo that will be constant for all the extensions added to this source tree are configured in `extensions/pom.xml` so that they do not need to be passed on the command line each time a new extension is added.
+
+[source,xml]
+----
+<plugin>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-maven-plugin</artifactId>
+    <version>${quarkus.version}</version>
+    <inherited>false</inherited>
+    <!-- Settings for stubbing new extensions via
+           mvn quarkus:create-extension -N -Dquarkus.artifactIdBase=my-new-extension
+    -->
+    <configuration>
+        <artifactIdPrefix>quarkus-</artifactIdPrefix>
+        <namePrefix xml:space="preserve">Quarkus - </namePrefix>
+    </configuration>
+</plugin>
+----
+
+TIP: The mojo derives artifact names from `artifactIdBase` passed on the command line.
+If you suspect that the derived name won't be what you want you can use `"-Dquarkus.nameBase=My Cool Extension`" on the command line.
```

Comment:
I can understand why `artifactIdBase` but I think it should be `name` not `nameBase`. It's the name of the extension.

Reply:
The mojo takes the definitions of `name` and `artifactId` from the Maven's `pom.xml` world. I believe that convention will be more intuitive to all users coming from outside.

`nameBase` is also not the full Maven module name. To make it a full name it needs to get prefixed with some common part, such as `Quarkus - `.